packages/apps/graph/src/graph/objects/Block.ts

```diff
@@ -17,23 +19,44 @@ export default builder.prismaNode('Block', {
 
     // relations
     transactions: t.prismaConnection({
+      args: {
+        events: t.arg.stringList({ required: false }),
+      },
       type: 'Transaction',
       cursor: 'block_requestkey',
-      totalCount(parent, args, context, info) {
+      async totalCount(parent, { events }, context, info) {
         return prismaClient.transaction.count({
           where: {
-            block: {
-              equals: parent.hash,
+            block: parent.hash,
+            requestkey: {
+              in: (
+                await prismaClient.$queryRaw<{ requestkey: string }[]>`
+                SELECT t.requestkey
+                FROM transactions t
+                INNER JOIN events e
+                ON e.block = t.block AND e.requestkey = t.requestkey
+                WHERE e.qualname IN (${Prisma.join(events as string[])})
+                AND t.block = ${parent.hash}`
+              ).map((r) => r.requestkey),
             },
           },
         });
       },
-      resolve(query, parent, args, context, info) {
+      async resolve(query, parent, { events }, context, info) {
         return prismaClient.transaction.findMany({
           ...query,
           where: {
-            block: {
-              equals: parent.hash,
+            block: parent.hash,
+            requestkey: {
+              in: (
+                await prismaClient.$queryRaw<{ requestkey: string }[]>`
```

Comment:
lgtm , just one small comment can't you abstract this yo a separate function, or and apply for line 33 and 52 since they are equal?

Reply:
Yes, will do.